ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py

```diff
@@ -29,110 +35,82 @@
 except ImportError:
     HAS_ANTA = False
 
-
 LOGGER = logging.getLogger(__name__)
 
 
-def _get_skipped_tests_from_tags(run_tags: tuple, skip_tags: tuple) -> list[dict]:
-    """
-    Arguments:
-      run_tags (tuple): Tuple of run_tags used to run the playbook.
-      skip_tags (tuple): Tuple of skip_tags used to run the playbook.
-
-    Returns:
-      result (list[dict]): List of dictionary containing the categories to skip
-    """
-    result = []
-    for cls, cls_info in AVD_TEST_CLASSES.items():
-        class_legacy_tags = set(cls_info.get("legacy_ansible_tags", {}))
-
-        if run_tags and "never" in class_legacy_tags:
-            other_tags = class_legacy_tags - {"never"}
-            if not other_tags.intersection(set(run_tags)):
-                result.append({"category": cls.__name__})
-                continue
-
-        if class_legacy_tags.intersection(set(skip_tags)):
-            result.append({"category": cls.__name__})
-            continue
-
-        if run_tags and run_tags != ("all",) and not class_legacy_tags.intersection(set(run_tags)):
-            result.append({"category": cls.__name__})
-
-    return result
-
-
 def get_anta_results(
     anta_device: AntaDevice,
     hostvars: Mapping,
     logging_level: str,
     skipped_tests: list[dict],
     ansible_tags: dict | None = None,
-    save_catalog_name: str | None = None,
-    dry_run: bool = False,
+    save_catalog_name: Path | None = None,
     yaml_dumper: Dumper | None = NoAliasDumper,
+    *,
+    dry_run: bool = False,
```

Comment:
why are you forcing this to be a kwarg?

Reply:
Calling a function with boolean positional arguments is confusing as the meaning of the boolean value is not clear to the caller.

Check `flake8-boolean-trap` for reference.